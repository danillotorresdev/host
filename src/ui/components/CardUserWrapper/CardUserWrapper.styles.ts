import { styled } from "@linaria/react";

export const SpinnerLoaderWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
`;