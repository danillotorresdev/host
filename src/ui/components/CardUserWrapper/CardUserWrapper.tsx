import { User } from "@/services/fetchUser/fetchUser.service";
import { SpinnerLoader } from "@/ui/components/Loaders/SpinnerLoader";
import { NotificationDisclaimer } from "@/ui/components/NotificationDisclaimer/NotificationDisclaimer";
import * as S from "./CardUserWrapper.styles";
import { CardUser } from "@/ui/components/CardUser/CardUser";

type CardUserWrapperProps = {
  username: string;
  isLoading: boolean;
  data: User | null | undefined;
  error: string | null;
};

export const CardUserWrapper = ({
  username,
  data,
  error,
  isLoading,
}: CardUserWrapperProps) => {
  if (error) {
    return <NotificationDisclaimer message={error} type="error" />;
  }

  if (isLoading) {
    return (
      <S.SpinnerLoaderWrapper>
        <SpinnerLoader size={50} />;
      </S.SpinnerLoaderWrapper>
    );
  }

  if (data === null) {
    return null;
  }

  return (
    <>
      {!data ? (
        <NotificationDisclaimer
          message={error ?? "UsuÃ¡rio nÃ£o encontrado"}
          type="error"
        />
      ) : (
        <CardUser
          username={username}
          name={data?.name}
          followers={data?.followers}
          following={data?.following}
          email={data?.email}
          bio={data?.bio}
          avatar_url={data?.avatar_url}
        />
      )}
    </>
  );
};
