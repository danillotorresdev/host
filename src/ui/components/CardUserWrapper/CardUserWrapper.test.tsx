import { CardUserWrapper } from "./CardUserWrapper";
import { render, screen } from "@/utils/settings/tests/utilities";
import { expect, it, describe } from "vitest";
import { BrowserRouter } from "react-router-dom";
import { userMock } from "@/services/fetchUser/user.mocks";

describe("CardUserWrapper", () => {
  it("should render the user's name", () => {
    const { asFragment } = render(
      <CardUserWrapper
        data={userMock}
        isLoading={false}
        error={null}
        username="jhondoe"
      />,
      {
        wrapper: BrowserRouter,
      },
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("John Doe")).toBeDefined();
  });
});
