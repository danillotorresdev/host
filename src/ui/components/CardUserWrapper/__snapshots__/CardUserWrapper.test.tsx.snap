// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardUserWrapper > should render the user's name 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-3 c1eelvmx"
  >
    <div
      class="mocked-styled-4 p1n234za"
    >
      <div
        class="mocked-styled-8 p1bhgu3m"
      >
        <div
          style="background-image: url(https://); background-repeat: no-repeat; background-position: center center; background-size: cover; filter: blur(20px); transition: filter 0.5s; width: 100%; height: 100%;"
        >
          <img
            alt=""
            src="https://"
            style="display: none;"
          />
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-7 tlp2jul"
    >
      <h2
        class="mocked-styled-5 ngisml4"
      >
        John Doe
      </h2>
      <div
        class="mocked-styled-6 pxniugy"
      >
        <svg
          class="svg svg-followers"
          data-testid="icon-svg"
          style="width: 24px; height: 24px; color: rgb(91, 91, 91);"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="/assets/sprite.svg#followers"
          />
        </svg>
        <p>
          Seguidores: 10
        </p>
      </div>
      <div
        class="mocked-styled-6 pxniugy"
      >
        <svg
          class="svg svg-followers"
          data-testid="icon-svg"
          style="width: 24px; height: 24px; color: rgb(91, 91, 91);"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="/assets/sprite.svg#followers"
          />
        </svg>
        <p>
          Seguindo: 10
        </p>
      </div>
      <div
        class="mocked-styled-6 pxniugy"
      >
        <svg
          class="svg svg-email"
          data-testid="icon-svg"
          style="width: 24px; height: 24px; color: rgb(91, 91, 91);"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="/assets/sprite.svg#email"
          />
        </svg>
        <p>
          johndoe@email.com
        </p>
      </div>
      <div
        class="mocked-styled-6 pxniugy"
      >
        <p>
          My bio
        </p>
      </div>
      <div
        style="width: 200px; margin-top: 1rem;"
      >
        <a
          class="mocked-styled-9 ltiwt9s"
          href="/user/jhondoe/repos"
          style="margin-bottom: 1rem;"
        >
          <svg
            class="svg svg-arrow-foward"
            data-testid="icon-svg"
            style="width: 24px; height: 24px; color: rgb(177, 174, 174);"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="/assets/sprite.svg#arrow-foward"
            />
          </svg>
           Reposit√≥rios
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
